core/src/main/scala/codesearch/core/db/repository/PackageDbRepository.scala

```diff
@@ -0,0 +1,44 @@
+package codesearch.core.db.repository
+
+import cats.Monad
+import doobie.Transactor
+import doobie.implicits._
+import fs2.Stream
+
+final case class Package(
+    name: String,
+    version: String
+)
+
+final case class PackageTableRow(
+    name: String,
+    version: String,
+    repository: String
+)
+
+trait PackageDbRepository[F[_]] {
+  def upsert(name: String, version: String, repository: String): F[Int]
+  def findByRepository(repository: String): Stream[F, Package]
+}
+
+object PackageDbRepository {
+  def apply[F[_]: Monad](xa: Transactor[F]): PackageDbRepository[F] = new PackageDbRepository[F] {
+    def upsert(name: String, version: String, repository: String): F[Int] = {
+      sql"""
+        INSERT INTO package(name, version, repository, updated_at)
+        VALUES ($name, $version, $repository, now())
```

Comment:
Will this escape `name`? Because otherwise we might get an SQL injection.

Reply:
Ok, I will check it.